lib/classes/PluginManager.js

```diff
@@ -309,6 +310,26 @@ class PluginManager {
     });
   }
 
+  loadVariableResolvers(pluginInstance) {
+    for (const [variablePrefix, resolverOrOptions] of _.entries(
+      pluginInstance.variableResolvers || {}
+    )) {
+      let options = resolverOrOptions;
+      if (typeof resolverOrOptions === 'function') {
+        options = {
+          resolver: resolverOrOptions,
+        };
+      }
+      this.serverless.variables.variableResolvers.push({
+        regex: new RegExp(`^${variablePrefix}:`),
+        method: options.resolver.bind(this.serverless.variables),
+        custom: true,
+        isDependentService: options.isDisabledAtPrepopulation,
+        name: options.serviceName,
```

Comment:
It'll be nice to have some validation here. so in case of unexpected structure, we do not throw cryptic errors as e.g. `undefined is not a function`.

- `variablePrefix` should pass some regex (I guess we should allow only alphanumeric chars)
- `resolver` should be a function
- In case of `options.isDisabledAtPrepopulation`, I would validate `serviceName`.

If we don't have a reliable utils for that, then [`type`](https://github.com/medikoo/type#type) package may be helpful. e.g.:

```javascript
const ensureFunction = require('type/plain-function/ensure');
const ensureString = require('type/string/ensure');

ensureFunction(options.resolver, {
  errorMessage: "expected 'resolver' (of variable resolver) to be a function, received %v"
});
ensureString(options.serviceName, {
  errorMessage: "'serviceName' (of variable resolver) is a mandatory option when `isDisabledAtPrepopulation` is set, received %v"
});
```